metalearners/_utils.py

```diff
@@ -487,3 +487,42 @@ def default_metric(predict_method: PredictMethod) -> str:
     if predict_method == _PREDICT_PROBA:
         return "neg_log_loss"
     return "neg_root_mean_squared_error"
+
+
+def check_onnx_installed():
+    """Checks that *onnx* is available."""
+    try:
+        import onnx  # noqa F401
+    except ImportError:
+        raise RuntimeError(
+            "onnx is not installed. Please install onnx to use this feature."
+        )
+
+
+def check_spox_installed():
+    """Checks that *spox* is available."""
+    try:
+        import spox  # noqa F401
+    except ImportError:
+        raise RuntimeError(
+            "spox is not installed. Please install spox to use this feature."
+        )
+
+
+def infer_dtype_and_shape_onnx(tensor):
```

Comment:
```suggestion
def infer_dtype_and_shape_onnx(tensor: onnx.TensorProto) -> tuple[np.dtype, tuple]:
```
Not sure about this, though -- you probably know better

Reply:
The return type is fine but the problem is with the parameter types. I don't want to import `onnx` at the start of the file so it's not a dependency but then I couldn't find a way to use typing and import the library inside the method.

Reply:
Ah, of course, I see!